config/config_test.go

```diff
@@ -915,6 +915,88 @@ func TestRequestLimitsValidateAndLog(t *testing.T) {
 	}
 }
 
+func TestRequestLogging(t *testing.T) {
+	hook := testLogrus.NewGlobal()
+
+	type logComponents struct {
+		msg string
+		lvl logrus.Level
+	}
+
+	testCases := []struct {
+		desc                string
+		inRequestLoggingCfg *RequestLogging
+		expectedLogInfo     []logComponents
+	}{
+		{
+			desc: "negative sampling rate, expect fatal log",
+			inRequestLoggingCfg: &RequestLogging{
+				RefererSamplingRate: -0.1,
+			},
+			expectedLogInfo: []logComponents{
+				{msg: `invalid config.request_logging.referer_sampling_rate: value must be positive and not greater than 1.0. Got -0.10`, lvl: logrus.FatalLevel},
+			},
+		},
+		{
+			desc: "Sampling rate greater than 1.0, expect fatal log",
+			inRequestLoggingCfg: &RequestLogging{
+				RefererSamplingRate: 1.1,
+			},
+			expectedLogInfo: []logComponents{
+				{msg: `invalid config.request_logging.referer_sampling_rate: value must be positive and not greater than 1.0. Got 1.10`, lvl: logrus.FatalLevel},
+			},
+		},
+		{
+			desc: "Sampling rate of 1.0 is between the acceptable threshold. Expect info log",
+			inRequestLoggingCfg: &RequestLogging{
+				RefererSamplingRate: 1.0,
+			},
+			expectedLogInfo: []logComponents{
+				{msg: `config.request_logging.referer_sampling_rate: 1.00`, lvl: logrus.InfoLevel},
+			},
+		},
+		{
+			desc: "Sampling rate of 0.0 is between the acceptable threshold. Expect info log",
+			inRequestLoggingCfg: &RequestLogging{
+				RefererSamplingRate: 0.0,
+			},
+			expectedLogInfo: []logComponents{
+				{msg: `config.request_logging.referer_sampling_rate: 0.00`, lvl: logrus.InfoLevel},
+			},
+		},
+		{
+			desc: "Sampling rate of 0.6814 is between the acceptable threshold. Expect info log",
+			inRequestLoggingCfg: &RequestLogging{
+				RefererSamplingRate: 0.6814,
+			},
+			expectedLogInfo: []logComponents{
+				{msg: `config.request_logging.referer_sampling_rate: 0.68`, lvl: logrus.InfoLevel},
+			},
+		},
+	}
+
+	//substitute logger exit function so execution doesn't get interrupted
+	defer func() { logrus.StandardLogger().ExitFunc = nil }()
+	logrus.StandardLogger().ExitFunc = func(int) {}
+
+	for _, tc := range testCases {
+		//Run
```

Comment:
Please wrap in `t.Run` following our latest pattern used in Prebid Server.

Reply:
Done